pkg/action/uninstall.go

```diff
@@ -70,9 +70,6 @@ func (u *Uninstall) Run(name string) (*release.UninstallReleaseResponse, error)
 	if err != nil {
 		return nil, errors.Wrapf(err, "uninstall: Release not loaded: %s", name)
 	}
-	if len(rels) < 1 {
-		return nil, errMissingRelease
```

Comment:
how does this removal affect the other storage drivers (secrets and configmaps)?

I'm concerned removing this will cause us to rely on the storage layer for this functionality, and there's no standard in the interface to capture the case when no releases were found.

Reply:
> how does this removal affect the other storage drivers (secrets and configmaps)?

Both the secrets and configmap drivers' implementation of `Query` already return `ErrReleaseNotFound` (see [secret driver](https://github.com/helm/helm/blob/release-3.1/pkg/storage/driver/secrets.go#L124) and [configmap driver](https://github.com/helm/helm/blob/release-3.1/pkg/storage/driver/cfgmaps.go#L133) respectively for the returned error) as is documented one level above in the [`History` storage](https://github.com/helm/helm/blob/release-3.1/pkg/storage/storage.go#L149) method, and so will already have returned two lines above.

> I'm concerned removing this will cause us to rely on the storage layer for this functionality, and there's no standard in the interface to capture the case when no releases were found.

Yep, I agree - but in that case, the places mentioned above in ConfigMap and Secret drivers which already return the error should be removed, `storage.History` updated to check and return the error explicitly, and this error here should be updated to have the correct type for consistency (`ErrReleaseNotFound` rather than `errMissingRelease`), I think? Our issue was simply that our test code using the `Memory` had different expectations of which error was returned compared to the production `Secret` driver.